internal/kgateway/extensions2/plugins/routepolicy/ai_policy.go

```diff
@@ -163,16 +163,21 @@ func applyDefaults(
 		}
 		var tmpl string
 		if field.Override != nil {
+			tmpl = string(marshalled)
+		} else {
 			// Inja default function will use the default value if the field provided is falsey
 			tmpl = fmt.Sprintf("{{ default(%s, %s) }}", field.Value, string(marshalled))
```

Comment:
also, this line should be:
```suggestion
			tmpl = fmt.Sprintf("{{ default(%s, %s) }}", field.GetField(), string(marshalled))
```